src/meta/typesmap.go

```diff
@@ -193,19 +244,42 @@ func (m TypesMap) Is(typ string) bool {
 		return false
 	}
 
-	_, ok := m.m[typ]
+	_, ok := m.m[NewType(typ)]
 	return ok
 }
 
+func (m TypesMap) AppendType(typ Type) TypesMap {
+	if !m.isImmutable() {
+		if m.m == nil {
+			m.m = make(RawTypesMap, 1)
+		}
+
+		m.m[typ] = struct{}{}
+
+		m.MarkAsImprecise()
+		return m
+	}
+
+	mm := make(RawTypesMap, 1)
+	for k, v := range m.m {
+		mm[k] = v
+	}
+
+	mm[typ] = struct{}{}
+
+	// The returned map is mutable and imprecise.
```

Comment:
Why is it imprecise? If you are appending the precise Type, how do you mark it as such?

Reply:
Initially, this is just a copy of the `AppendString` function modified for types, in `AppendString` we consider that the map is mutable and imprecise, so this has lagged behind in this function as well.